packages/host/app/services/billing-service.ts

```diff
@@ -0,0 +1,14 @@
+import Service from '@ember/service';
+
+import { getMatrixUsername } from '@cardstack/runtime-common/matrix-client';
+
+import ENV from '@cardstack/host/config/environment';
+const { stripePaymentLink } = ENV;
+
+export default class BillingService extends Service {
+  getStripePaymentLink(matrixUserId: string): string {
+    const username = getMatrixUsername(matrixUserId);
```

Comment:
`matrix_user_id` in `users` table will be saved in the form of `@username:homeserver.tld`. I think you don't want to discard parts of the id otherwise the [stripe handler]( https://github.dev/cardstack/boxel/blob/db213a41148bd19b8b39f802f18b7cb320a54306/packages/billing/stripe-webhook-handlers/checkout-session-completed.ts#L33-L34) will fail to find the user that needs to be updated with the stripe customer id.

Maybe you can base64 the matrix user id (in order to produce alphanumeric-only) and set that value as `clientReferenceId`. Alternatively, you could use the user's UUID, and use that as `clientReferenceId`. Probably you'd need to query the recently added `user` endpoint to get that UUID. If you go this route then don't forget to fix the webhook handler to identify the user correctly. 

Moreover, I think if you want to properly test this mechanism and make sure the Stripe's free plan payment link properly associates the user with their stripe id, it would be a good idea to test using a real Stripe environment. I can add you to our sandbox environment and you'll be able to try it out

Reply:
Also add a test that the free plan's stripe payment link is correct (especially the `client_reference_id` part) - not sure you already have that but I think I didn't see it

Reply:
Thanks for the suggestion @jurgenwerk , I will try it out - let me know if you grant access to our sandbox environment too

There is [a blocker](https://discord.com/channels/584043165066199050/900021750367129641/1306326428090110105) to add the test cases in Matrix so far, I will update the PR again once it's resolved